packages/legend-graph/src/graph/metamodel/pure/packageableElements/domain/Multiplicity.ts

```diff
@@ -18,8 +18,8 @@ import { CORE_HASH_STRUCTURE } from '../../../../../graph/Core_HashUtils.js';
 import { hashArray, type Hashable } from '@finos/legend-shared';
 
 export class Multiplicity implements Hashable {
-  lowerBound: number;
-  upperBound?: number | undefined;
+  readonly lowerBound: number;
```

Comment:
did you check the `...Observer` stuff to see if we decorate these as `:observable`? since `readonly` fields should not be `observable`, hence the `hashCode` for multiplicity probably don't need to be `computed` as well